evalml/data_checks/outliers_data_check.py

```diff
@@ -42,17 +41,18 @@ def validate(self, X, y=None):
 
         X = infer_feature_types(X)
         X = X.ww.select("numeric")
+        X.ww.drop(X.columns[X.isna().all()].tolist(), inplace=True)
 
         if len(X.columns) == 0:
             return results
 
         has_outliers = []
         for col in X.columns:
-            outlier_results = OutliersDataCheck._outlier_score(X[col], False)
-            if (
-                outlier_results is not None and outlier_results["score"] <= 0.9
```

Comment:
Woodwork doesn't perform this kind of outlier score calculation, but it does return the necessary information to calculate that, if you still only want to add the column when it's below `0.9`

```python
pct_outliers = (len(box_plot_dict["low_values"]) + len(box_plot_dict["high_values"])) / len(column_nonan)

num_records = len(column_nonan)
score = OutliersDataCheck._no_outlier_prob(num_records, pct_outliers)
```